package.json

```diff
@@ -18,27 +18,30 @@
     "sideEffects": false,
     "exports": {
         ".": {
-            "import": "./index.mjs",
-            "require": "./index.js"
-        },
-        "./package.json": "./package.json"
+            "types": "./index.d.ts",
+            "default": "./index.js"
+        }
     },
-    "main": "index",
-    "module": "index.mjs",
+    "main": "index.js",
+    "types": "index.d.ts",
     "files": [
-        "index.*"
+        "index.d.ts",
+        "index.js"
     ],
     "scripts": {
-        "prebuild": "npm run -s clean",
-        "build": "rollup -c",
-        "clean": "rimraf .nyc_output coverage index.*",
+        "prebuild": "npm run -s clean ",
+        "build": "tsc -p declaration.tsconfig.json && rollup -c",
+        "clean": "rimraf .nyc_output coverage dist index.*",
         "coverage": "opener ./coverage/lcov-report/index.html",
         "docs:build": "vitepress build docs",
         "docs:watch": "vitepress dev docs",
         "format": "npm run -s format:prettier -- --write",
         "format:prettier": "prettier .",
         "format:check": "npm run -s format:prettier -- --check",
-        "lint": "eslint .",
+        "lint:eslint": "eslint .",
+        "lint:tsc": "tsc",
```

Comment:
Naming: I wouldn't call TypeScript a "lint" task. Even if it acts essentially as a linter in this setup, I've found it confusing for newer contributors to see it named as one.